tests/testthat/test-plot_seromodel.R

```diff
@@ -0,0 +1,313 @@
+library(testthat)
+library(ggplot2)
+library(rlang)
+library(purrr)
+
+# NOTE: If the variable `expected_plot` needs to be updated because of changes in the plots
+# You can use dput(extract_plot_data(plot)) to obtain the current structure of the plot
+
+# Common data
+data(veev2012)
+serosurvey <- veev2012
+
+
+create_prepared_serosurvey <- function(actual_serosurvey) {
+  return(prepare_serosurvey_for_plotting(actual_serosurvey %>%
+    add_age_group_to_serosurvey()))
+}
+
+
+test_that("plot_serosurvey creates a ggplot with correct structure", {
+  skip_on_cran()
+
+  seromodel <- fit_seromodel(
+    serosurvey = serosurvey
+  )
+  actual_plot <- extract_plot_data(plot_serosurvey(serosurvey))
+
+  prepared_serosurvey <- create_prepared_serosurvey(serosurvey)
+
+  expected_plot <- list(
+    classes = c("gg", "ggplot"),
+    layers = list(
+      list(
+        mapping = list(
+          ymin = "seroprev_lower", ymax = "seroprev_upper"
+        ), geom_params = list(
+          na.rm = FALSE, orientation = NA, width = 0.1
+        )
+      ),
+      list(
+        mapping = list(
+          y = "seroprev", size = "n_sample"
+        ), geom_params = list(na.rm = FALSE)
+      )
+    ),
+    coordinates = list(limits = list(x = c(1, 60), y = c(
```

Comment:
I suggest ussing `x = c(min(serosurvey$age_min), max(serosurvey$age_max))` instead of `x = c(1, 60)`, just in case that `veev2012` were to change or be replaced in the test for a different dataset. This is in line with the inner working of the function as well (see https://github.com/epiverse-trace/serofoi/blob/72b6c7fe974549f95e1063ef2379b8bea39bf723/R/plot_seromodel.R#L172-L175).